### ISSUE TITLE
Serverless Deploy failing

### ISSUE THREAD

**Issue by atharrison** [CircleCI-Public/serverless-framework-orb#51]
**Orb Version**
`circleci/serverless-framework@2.0.1`

**Describe the bug**
The "Install Serverless CLI" step is failing, following the instructions from https://circleci.com/developer/orbs/orb/circleci/serverless-framework#orb-source

**To Reproduce**

Run a serverless deploy in the CircleCI pipeline.

Our deploy config, essentially straight from the CircleCI documentation:
```
  deploy-prod:
    executor: serverless-framework/default
    steps:
      - attach_workspace:
          at: .
      - aws-cli/setup
      - serverless-framework/setup
      - run:
          command: serverless deploy --stage prod --verbose
          name: deploy
```

**Expected behavior**

The serverless deploy executes.

**Current behavior**
<img width="958" alt="Screenshot 2024-05-21 at 1 43 31 PM" src="https://github.com/CircleCI-Public/serverless-framework-orb/assets/1900008/f8ae92bd-0030-470d-b4cb-699f2589d77c">

**Additional context**

This was working yesterday, but started failing today. I've been informed that the Serverless team released their v4 beta today, so there may be something going on there.

Link to Serverless thread: https://forum.serverless.com/t/serverless-deploy-failing-when-run-from-circleci/20184

**Comment by brunodamacena97** [comment]
+1, it's happening to me as well.

**Comment by geminiDelta** [comment]
+1, same here
seems to have been broken by this commit https://github.com/serverless/serverless/commit/bb55e0b38c30348c3284bbdba672b97cf18f3d39
which removed the `scripts/pkg/install.sh` script referenced by this orb [here](https://github.com/CircleCI-Public/serverless-framework-orb/blob/cfd5d423f61b2cec0ed875b2900fc6e62e1d09b7/src/scripts/install.sh#L7C13-L7C36) (`https://slss.io/install` redirects to `https://raw.githubusercontent.com/serverless/serverless/main/scripts/pkg/install.sh`).

**Comment by elliott-chenger-ribbiot** [comment]
In our forum post with the Serverless team, they have suggested that they are going to fix the broken link. Will keep this thread posted once we see it working again on our side.

If this doesn't work then I think we potentially try and PR a small fix to hard code the exact location for now and then work with their team to figure out a better long term strategy. Thoughts?

**Comment by geminiDelta** [comment]
Sounds good. If it does come to temporarily hard coding a location, I would recommend one of these:
- https://raw.githubusercontent.com/serverless/serverless/v3.38.0/scripts/pkg/install.sh is the last tagged version
- https://sls-standalone-sv-1300963013.cos.na-siliconvalley.myqcloud.com/install.sh was used in v2.0.0 of this orb

**Comment by elliott-chenger-ribbiot** [comment]
Thanks @geminiDelta for the suggestions.

Yeah likely to make it so that none of us have any other weird issues, I would probably just point it to the 2.0.0 if that was, what was used to approve this orb. I know technically the shortlink allowed them to bypass any "test" process. But it may just be more tolerable for us all until we can figure out the long term flow here.

**Comment by elliott-chenger-ribbiot** [comment]
@KyleTryon or @EricRibeiro how would you like me to contribute this change? I can fork and have my PR or can y'all add me to this project so I can push up my branch?

**Comment by geminiDelta** [comment]
Just had a thought regarding the link used for [v2.0.0](https://github.com/CircleCI-Public/serverless-framework-orb/blob/e9fe96f75f8b0c6b8edaddd0be20a91bcae7778c/src/scripts/install.sh#L2C24-L2C105). It looks like a CDN, which could start to fail as well depending on where it's sourcing the file from (e.g. `main` branch where it's been deleted) and how cache invalidation, etc is configured. You would probably know more than I do about that, but I wonder if the [tagged version](https://raw.githubusercontent.com/serverless/serverless/v3.38.0/scripts/pkg/install.sh) in GitHub could be considered more reliable/permanent ([link to repo for ref](https://github.com/serverless/serverless/blob/v3.38.0/scripts/pkg/install.sh)).

**Comment by EricRibeiro** [comment]
Neither Kyle nor I are with CircleCI anymore. I suggest contacting their support team to get some insight into this issue.

**Comment by omi-jbrivero** [comment]
+ 1 on this issue :)

**Comment by salemgolemugoo** [comment]
If you guys need a quick fix, i've published a new version to my circleci registry. You can use it like this:
```
version: 2.1

orbs:
  node: circleci/node@5.1.0
  serverless-framework: salemgolemugoo/serverless-framework@dev:issue-51

jobs:
  deploy:
    docker:
      - image: cimg/base:2023.06
    steps:
      - checkout
      - node/install:
          node-version: '14.21.3'
      - serverless-framework/setup
```

> [!CAUTION]
> You need to disable "Orb Security Settings" in "Organization Settings -> Security"

**Comment by elliott-chenger-ribbiot** [comment]
It looks like CircleCI has patched and released a fix for us. 🎉 https://github.com/CircleCI-Public/serverless-framework-orb/pull/54

We can probably close this out. However we still need to make sure that there is a better long term solution for this. Will see if I can follow up with both groups to understand what that looks like.

**Comment by jgrugru** [comment]
+1 Also happening to us

**Comment by atharrison** [comment]
Confirmed that this is working again for us. For those scrolling quickly to the bottom:

Update your orb:
```
orbs:
  serverless-framework: circleci/serverless-framework@2.0.2
```

**Comment by zaki-arain** [comment]
Thanks for your patience while we looked into this. We'll close this issue as resolved by #54 😄
